src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferReadOnlySpanPropertiesOverReadOnlyArrayFields.cs

```diff
@@ -0,0 +1,493 @@
+                            fieldReference.Parent.Accept(fieldReferenceVisitor, new VisitContext(fieldReference, fieldReference.Field, context.CancellationToken));
+                        }
+                        break;
+                    case IFieldInitializerOperation fieldInitializer:
+
+                        //  Eliminate candidates without an all-constant array initializer
+                        if (fieldInitializer.Value is IArrayCreationOperation arrayCreation &&
+                            arrayCreation.Initializer is not null &&
+                            arrayCreation.Initializer.ElementValues.All(x => x.ConstantValue.HasValue))
+                        {
+                            foreach (var field in fieldInitializer.InitializedFields)
+                                cache.AddFieldWithValidFieldInitializer(field);
+                        }
+                        break;
+                }
+            }
+
+            //  Report diagnostics for all fields that survived candidate elimination and have a valid field initializer.
+            void OnCompilationEnd(CompilationAnalysisContext context)
+            {
+                var asSpanInvocationLookup = cache.GetSavedOperationsLookup();
+                foreach (var field in cache.CandidatesWithValidFieldInitializers)
+                {
+                    //  Save the locations of all operations that need to be fixed by the fixer.
+                    var savedLocations = asSpanInvocationLookup[field].Select(x => new SavedSpanLocation(x.Syntax.Span, x.Syntax.SyntaxTree.FilePath));
+                    string propertyValue = SavedSpanLocation.Serialize(savedLocations);
+                    var properties = ImmutableDictionary<string, string?>.Empty.Add(FixerDataPropertyName, propertyValue);
+                    var messageArgument = ((IArrayTypeSymbol)field.Type).ElementType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
+                    var diagnostic = field.CreateDiagnostic(Rule, properties, messageArgument);
+                    context.ReportDiagnostic(diagnostic);
+                }
+                cache.Dispose();
+            }
+        }
+
+        //  Not compared for equality
+#pragma warning disable CA1815
+        private readonly struct VisitContext
+#pragma warning restore CA1815
+        {
+            public VisitContext(IOperation operation, IFieldSymbol field, CancellationToken cancellationToken)
+            {
+                Operation = operation;
+                Field = field;
+                CancellationToken = cancellationToken;
+            }
+
+            public IOperation Operation { get; }
+            public IFieldSymbol Field { get; }
+            public CancellationToken CancellationToken { get; }
+
+            public VisitContext WithOperation(IOperation newOperation) => new(newOperation, Field, CancellationToken);
+        }
+
+        //  Not compared for equality
+#pragma warning disable CA1815
```

Comment:
Seems it is checking for equality https://github.com/dotnet/roslyn-analyzers/blob/5ad4dd91db2fed2942908b7edb941c21a303bfa7/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferReadOnlySpanPropertiesOverReadOnlyArrayFields.cs#L216-L220
```suggestion
```